p2p/net/upgrader/upgrader.go

```diff
@@ -257,17 +263,18 @@ func (u *upgrader) setupMuxer(ctx context.Context, conn sec.SecureConn, server b
 		if err != nil {
 			return
 		}
+		muxerID = m.ID
 		smconn, err = m.Muxer.NewConn(conn, server, scope)
 	}()
 
 	select {
 	case <-done:
```

Comment:
Minor nit, I think it would be better to return the value here (a la message passing) rather than mutating shared state.

Reply:
I think I agree. Changed.